src/Land/Model/LandModel.jl

```diff
@@ -205,7 +222,16 @@ function source!(
     t::Real,
     direction,
 )
-    land_source!(land.source, land, source, state, diffusive, aux, t, direction)
+    tend = Source()
+    _args = (; state, aux, t, direction, diffusive)
+    args = merge(_args, (precomputed = precompute(land, _args, tend),))
+
+    map(prognostic_vars(land)) do prog
```

Comment:
FWIW, this is a little opaque (though the comments in the src file for `get_prog_state` help a lot)...
we have `prog` referring to e.g. an instance of type  of Prognostic variable, e.g. InternalEnergy(), 
`var` referring to `state`, `source`, or `flux`, etc.
`name` something else
as a physicist, I would say the variable is rhoe_int, and that would be it. so maybe this is some of the "under the hood" code leaking through?


In any case, I think this accumulates all sources for that variable and then sets source.soil.heat.rho_eint = val, and someone adding in a new source or flux term probably doesnt need to parse it.

Reply:
Yeah, this is pretty generic code. You'll see that the Atmos implementation is exactly the same with `atmos` instead of `land`. This could eventually be moved to `BalanceLaws`.

> In any case, I think this accumulates all sources for that variable and then sets source.soil.heat.rho_eint = val, and someone adding in a new source or flux term probably doesnt need to parse it.

Correct.